spec.html

```diff
@@ -344,115 +269,165 @@ <h1>
+            </th>
+          </tr>
+          <tr>
+            <td>
+              [[AsyncSnapshotMapping]]
+            </td>
+            <td>
+              a List of Async Context Mapping Records
+            </td>
+            <td>
+              Represents the snapshotted surrounding agent's Agent Record's [[AsyncContextMapping]] of the AsyncSnapshot instance.
+            </td>
+          </tr>
+        </table>
+      </emu-table>
+    </emu-clause>
+  </emu-clause>
+
+  <emu-clause id="sec-asynclocal-objects">
+    <h1>AsyncLocal Objects</h1>
+
+    <emu-clause id="sec-asynclocal-constructor">
+      <h1>The AsyncLocal Constructor</h1>
+      <p>The AsyncLocal constructor:</p>
+      <ul>
+        <li>is <dfn>%AsyncLocal%</dfn>.</li>
+        <li>is the initial value of the *"AsyncLocal"* property of the global object.</li>
+        <li>creates and initializes a new AsyncLocal when called as a constructor.</li>
+        <li>is not intended to be called as a function and will throw an exception when called in that manner.</li>
+        <li>may be used as the value in an `extends` clause of a class definition. Subclass constructors that intend to inherit the specified AsyncLocal behaviour must include a `super` call to the AsyncLocal constructor to create and initialize the subclass instance with the internal state necessary to support the `AsyncLocal` and `AsyncLocal.prototype` built-in methods.</li>
+      </ul>
+
+      <emu-clause id="sec-asynclocal">
+        <h1>AsyncLocal ( _options_ )</h1>
+        <p>This function performs the following steps when called:</p>
 
         <emu-alg>
-          1. Let _snapshot_ be AsyncContextSnapshot().
-          1. If IsCallable(_target_) is false, throw a TypeError exception.
-          1. Let _F_ be ? AsyncContextWrappedFunctionCreate(_target_, _snapshot_).
-          1. Let _L_ be 0.
-          1. Let _targetHasLength_ be ? HasOwnProperty(_target_, *"length"*).
-          1. If _targetHasLength_ is *true*, then
-            1. Let _targetLen_ be ? Get(_target_, *"length"*).
-            1. If _targetLen_ is a Number, then
-              1. If _targetLen_ is *+∞*<sub>𝔽</sub>, set _L_ to +∞.
-              1. Else if _targetLen_ is *-∞*<sub>𝔽</sub>, set _L_ to 0.
-              1. Else,
-                1. Let _targetLenAsInt_ be ! ToIntegerOrInfinity(_targetLen_).
-                1. Assert: _targetLenAsInt_ is finite.
-                1. Set _L_ to _targetLenAsInt_.
-          1. Perform SetFunctionLength(_F_, _L_).
-          1. Let _targetName_ be ? Get(_target_, *"name"*).
-          1. If _targetName_ is not a String, set _targetName_ to the empty String.
-          1. Perform SetFunctionName(_F_, _targetName_, *"wrapped"*).
-          1. Return _F_.
+          1. If NewTarget is *undefined*, throw a *TypeError* exception.
+          1. If _options_ is an Object, then
+            1. Let _name_ be ? Get(_options_, *"name"*).
+            1. Let _nameStr_ be ? ToString(_name_).
```

Comment:
This will result in `new AsyncLocal({}).name === "undefined"`. What we probably want is to only set name if `Has(options, "name")`.

Reply:
Thanks, fixed!